quinn-proto/src/connection.rs

```diff
@@ -57,6 +57,10 @@ pub struct Connection {
     /// Limit on incoming data
     pub local_max_data: u64,
     client_config: Option<ClientConfig>,
+    /// Incoming cryptographic handshake stream
+    crypto_stream: stream::Assembler,
```

Comment:
So I don't like having the `Assembler` here much. It seems like a layering violation in that `Assembler` should be an implementation detail of `Recv`. Also, don't we actually need more-or-less a full `(Send, Recv)` here? That would also cover the separate `crypto_offset` below. The crypto stream still has to deal with ~all of the issues that another stream would have to deal with, right?

Reply:
The crypto stream is not subject to finishing, resets, stops, or flow control, and does not benefit from unordered reads. As such, everything but a stream assembler and a send offset (i.e. the vast majority of the `Send` and `Recv` structs) would be vestigal.

Reply:
Alright, in that case it would be overkill.